chrome/content/zotero/xpcom/zotero.js

```diff
@@ -1069,53 +1088,44 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 				fp.appendFilters(nsIFilePicker.filterAll);
 				if (fp.show() == nsIFilePicker.returnOK) {
 					var file = fp.file;
-					
-					if (file.directoryEntries.hasMoreElements()) {
+					let dialogText = '';
+					let dialogTitle = '';
+
+					// In dropbox folder
+					if (Zotero.File.isDropboxDirectory(file.path)){
+						dialogTitle = Zotero.getString('dataDir.selectedDirInDropbox.title');
+						dialogText = Zotero.getString('dataDir.selectedDirInDropbox.text');
+					}
+					else if (file.directoryEntries.hasMoreElements()) {
 						var dbfile = file.clone();
 						dbfile.append('zotero.sqlite');
 						
 						// Warn if non-empty and no zotero.sqlite
 						if (!dbfile.exists()) {
-							var buttonFlags = ps.STD_YES_NO_BUTTONS;
-							if (moreInfoCallback) {
-								buttonFlags += ps.BUTTON_POS_2 * ps.BUTTON_TITLE_IS_STRING;
-							}
-							var index = ps.confirmEx(null,
-								Zotero.getString('dataDir.selectedDirNonEmpty.title'),
-								Zotero.getString('dataDir.selectedDirNonEmpty.text'),
-								buttonFlags,
-								null,
-								null,
-								moreInfoCallback ? Zotero.getString('general.help') : null,
-								null, {});
-							
-							// Not OK -- return to file picker
-							if (index == 1) {
-								continue;
-							}
-							else if (index == 2) {
-								setTimeout(function () {
-									moreInfoCallback();
-								}, 1);
-								return false;
-							}
+							dialogTitle = Zotero.getString('dataDir.selectedDirNonEmpty.title');
+							dialogText = Zotero.getString('dataDir.selectedDirNonEmpty.text');
 						}
 					}
-					else {
+					else { // Directory empty
+						dialogTitle = Zotero.getString('dataDir.selectedDirEmpty.title');
+						dialogText = Zotero.getString('dataDir.selectedDirEmpty.text', Zotero.appName) + '\n\n'
+								+ Zotero.getString('dataDir.selectedDirEmpty.useNewDir');
```

Comment:
Yeah, this is old code, so we can discuss that separately. (It'd be great to do, but we didn't do it in the past because we don't have a native method for moving a directory, and it's somewhat of a scary proposition — anytime we touch files we end up with user reports because of security software and the like. At the very least, we'd need a function that did a whole lot of tracking and cleanup on errors. We also need to shut down everything beforehand and be prepared to start it up again after recovering from an error, though that part, at least, we now have the ability to do.)